pkgs/applications/graphics/vpuppr-bin/default.nix

```diff
@@ -0,0 +1,111 @@
+{ lib
+, stdenv
+, alsa-lib
+, gcc-unwrapped
+, git
+, libGLU
+, libglvnd
+, libpulseaudio
+, perl
+, zlib
+, udev
+, libX11
+, libXcursor
+, libXext
+, libXfixes
+, libXi
+, libXinerama
+, libXrandr
+, libXrender
+, fetchzip
+, autoPatchelfHook
+, copyDesktopItems
+, makeDesktopItem
+, fetchurl
+, makeWrapper
+}:
+
+stdenv.mkDerivation rec {
+  pname = "vpuppr-bin";
+  version = "0.9.0";
+
+  src = fetchzip {
+    url = "https://github.com/virtual-puppet-project/vpuppr/releases/download/${version}/vpuppr_${version}_linux.zip";
+    hash = "sha256-JKiVM1N/NOa5StqUG+BV7ZvNZmDfi5jNKaeWJLy/pTA=";
+  };
+
+  icon = fetchurl {
+    url = "https://github.com/virtual-puppet-project/vpuppr/raw/${version}/assets/osfgd_icon.png";
+    hash = "sha256-2SdiUCfHLhbMKzpfyulp/qDGShNwCvMXotkBH0I7VDM=";
+  };
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Virtual Puppet Project";
+      desktopName = "Virtual Puppet Project";
+      comment = "VTuber application made with Godot 3.5";
+      exec = "vpuppr";
+      icon = "vpuppr_icon";
+      categories = [ "Graphics" "AudioVideo" "Recorder" ];
+      terminal = false;
+    })
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+    makeWrapper
+  ];
+
+  buildInputs = [
+    alsa-lib
+    gcc-unwrapped.lib
+    git
+    libGLU
+    libglvnd
+    libpulseaudio
+    perl
+    zlib
+    udev
+    libX11
+    libXcursor
+    libXext
+    libXfixes
+    libXi
+    libXinerama
+    libXrandr
+    libXrender
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    # vpuppr requires its resources to be in the same directory as the executable
+    # hence we place everything in share first and symlink the executable to bin
+
+    mkdir -p $out/share/vpuppr
+    cp -r resources $out/share/vpuppr/
+    cp vpuppr.pck $out/share/vpuppr/
+    install -Dm555 vpuppr.x86_64 $out/share/vpuppr/vpuppr.x86_64
+
+    mkdir -p $out/bin
+    ln -s $out/share/vpuppr/vpuppr.x86_64 $out/bin/vpuppr
+
+    wrapProgram $out/bin/vpuppr \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath buildInputs}"
```

Comment:
Can we avoid LD_LIBRARY_PATH by using patchelf/autoPatchelfHook?

Reply:
I have attempted to get this to work with patchelf/autoPatchelfHook - as per https://github.com/NixOS/nixpkgs/blob/7a7b0862dbe772582eb3e41c85430faf206fb606/pkgs/games/oh-my-git/default.nix#L107-L111

Unfortunately this doesn't appear to work - it looks like it's loading libraries at runtime and despite doing the equivalent of oh-my-git the program is still unable to find libudev.so.1.

Reply:
To note, the error I'm getting:

```
Godot Engine v3.5.1.rc.custom_build.293c3844b - https://godotengine.org
WARNING: Error setting locale modifiers
     at: initialize (platform/x11/os_x11.cpp:166)
WARNING: XOpenIM failed
     at: initialize (platform/x11/os_x11.cpp:213)
WARNING: XCreateIC couldn't create xic
     at: initialize (platform/x11/os_x11.cpp:510)
OpenGL ES 3.0 Renderer: NVIDIA GeForce RTX 2080 SUPER/PCIe/SSE2
Async. shader compilation: OFF
WARNING: All audio drivers failed, falling back to the dummy driver.
     at: initialize (servers/audio_server.cpp:222)
libudev.so.1: cannot open shared object file: No such file or directory
```

Reply:
Can you fix this by hand with patchelf and add a rpath for libudev.so.1?

Reply:
I tried as per the posts above - but my nix knowledge appears to not be good enough to try and solve why it's failing... any thoughts?